File: 1. Two Sum/C#/two_sum.cs
public class Solution {
  public int[] TwoSum(int[] nums, int target) {
      Dictionary<int, int> numMap = new Dictionary<int, int>();
      
      for (int i = 0; i < nums.Length; i++)
      {
        int num = nums[i];
        int diff = target - num;
        
        if (numMap.ContainsKey(diff))
          return new int[] {i, (int)numMap[diff]};

        if (!numMap.ContainsKey(num))
          numMap.Add(num, i);
      }
      
      return null;
  }
}

File: 1. Two Sum/TypeScript/two_sum.ts
function twoSum(nums: number[], target: number): number[] {
    let numMap = {};
    let returnable = [];
    nums.forEach((num, i) => {
        const complement = target - num
        if (numMap[complement] !== undefined) { returnable = [numMap[complement], i] }
        numMap[nums[i]] = i
    })
    return returnable
};